src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -1757,68 +1758,97 @@ private void handlePostBlob(HttpServletRequest request,
-                        "Content-Disposition: form-data;" +
-                        " name=\"acl\"")) {
+                if (isField(header, "acl")) {
                     // TODO: acl
-                } else if (startsWithIgnoreCase(header,
-                        "Content-Disposition: form-data;" +
-                        " name=\"AWSAccessKeyId\"")) {
+                } else if (isField(header, "AWSAccessKeyId") ||
+                        isField(header, "X-Amz-Credential")) {
                     identity = new String(baos.toByteArray());
-                } else if (startsWithIgnoreCase(header,
-                        "Content-Disposition: form-data;" +
-                        " name=\"Content-Type\"")) {
+                } else if (isField(header, "Content-Type")) {
                     contentType = new String(baos.toByteArray());
-                } else if (startsWithIgnoreCase(header,
-                        "Content-Disposition: form-data;" +
-                        " name=\"file\"")) {
+                } else if (isField(header, "file")) {
                     // TODO: buffers entire payload
                     payload = baos.toByteArray();
-                } else if (startsWithIgnoreCase(header,
-                        "Content-Disposition: form-data;" +
-                        " name=\"key\"")) {
+                } else if (isField(header, "key")) {
                     blobName = new String(baos.toByteArray());
-                } else if (startsWithIgnoreCase(header,
-                        "Content-Disposition: form-data;" +
-                        " name=\"policy\"")) {
+                } else if (isField(header, "policy")) {
                     policy = baos.toByteArray();
-                } else if (startsWithIgnoreCase(header,
-                        "Content-Disposition: form-data;" +
-                        " name=\"signature\"")) {
+                } else if (isField(header, "signature") ||
+                        isField(header, "X-Amz-Signature")) {
                     signature = new String(baos.toByteArray());
+                } else if (isField(header, "X-Amz-Algorithm")) {
+                    algorithm = new String(baos.toByteArray());
                 }
             }
             nextPart = multipartStream.readBoundary();
         }
 
-        if (identity == null || signature == null || blobName == null ||
-                policy == null) {
+
+        if (blobName == null || policy == null) {
             response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
             return;
         }
 
-        Map.Entry<String, BlobStore> provider =
-                blobStoreLocator.locateBlobStore(identity, null, null);
-        if (provider == null) {
-            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+        String headerAuthorization = null;
+        S3AuthorizationHeader authHeader = null;
+        boolean signatureVersion4;
+        if (algorithm == null) {
```

Comment:
I don't quiet understand what you mean by this. Can you point me to the lines in the code where such a check is performed?